arbor/mc_cell_group.cpp

```diff
@@ -406,15 +408,40 @@ void mc_cell_group::advance(epoch ep, time_type dt, const event_lane_subrange& e
             call_info.push_back({sa.sampler, pid, p.tag, n_samples, n_samples+n_times*n_raw});
 
             for (auto t: sample_times) {
+                auto intdom = cell_to_intdom_[cell_index];
                 for (probe_handle h: p.handle.raw_handle_range()) {
-                    sample_event ev{t, (cell_gid_type)cell_to_intdom_[cell_index], {h, n_samples++}};
+                    sample_event ev{t, (cell_gid_type)intdom, {h, n_samples++}};
                     sample_events.push_back(ev);
                 }
+                if (sa.policy==sampling_policy::exact) {
+                    target_handle h(-1, 0, intdom);
```

Comment:
Is the `-1` index used to indicate that the event is not to be delivered to a target?

Reply:
Yes: -1 is guaranteed not to be a valid mechanism id.